### PR TITLE
Fixes ruins air alarms showing up on station tablets

### LINKED ISSUE #38462: Tablet alarm monitor shows non-station alarms
[Round ID]: 89408

[Reproduction]: Open up the tablet, download the alarm monitor. Open it up and wait for stuff to happen on syndie lavaland outposts or similar non-station facilities. Or go cause an alarm. 

In the case of round 89408, SYndicate Lavaland Experimentation Lab fire alarms showed up at around 01:08:00 round time. Have seen comms outpost atmos alarms pop up in previous rounds.
Also worth mentioning the on-station alarm monitor terminal does not record these alarms.

### PR DESCRIPTION
**Pull request by CitrusGender**
[Changelogs]: Code had a check for station level air alarms and mining alarms but didn't think that ruins would have them. Simple fix. Edit: Also added myself to the config for the coder server.
:cl:
fix: Ruins air alarms will no longer be reported in the station tablets
/:cl:

[why]: fixes #38462

### CODE REVIEW

File: code/modules/modular_computers/file_system/programs/alarm.dm [hunk 1]
```diff
@@ -40,7 +40,7 @@
 	return data
 
 /datum/computer_file/program/alarm_monitor/proc/triggerAlarm(class, area/A, O, obj/source)
-	if(!is_station_level(source.z) && !is_mining_level(source.z))
+	if((!is_station_level(source.z) && !is_mining_level(source.z)) || istype(A, /area/ruin))
```
Comment (ShizCalev) [inline code/modules/modular_computers/file_system/programs/alarm.dm]: check it against `GLOB.the_station_areas` instead imo

Comment (CitrusGender) [reply code/modules/modular_computers/file_system/programs/alarm.dm]: didn't even realize that was a thing, sure.

**Comment by CitrusGender** [comment]
@ShizCalev They don't get mining level air alarms now. I guess that's acceptable (but not intended) since technically they use servers on the station.
https://github.com/tgstation/tgstation/blob/8e692e6c02786b7b8e605b942d1568af5387abed/code/controllers/subsystem/mapping.dm#L207

File: code/modules/modular_computers/file_system/programs/alarm.dm [hunk 2]
```diff
@@ -40,7 +40,7 @@
 	return data
 
 /datum/computer_file/program/alarm_monitor/proc/triggerAlarm(class, area/A, O, obj/source)
-	if(!is_station_level(source.z) && !is_mining_level(source.z))
+	if((!is_station_level(source.z) && !is_mining_level(source.z)) || !(A.type in GLOB.the_station_areas))
```
Comment (Jalleo) [inline code/modules/modular_computers/file_system/programs/alarm.dm]: Will only generate on the main z level
https://github.com/tgstation/tgstation/blob/8e692e6c02786b7b8e605b942d1568af5387abed/code/controllers/subsystem/mapping.dm#L209 This line only runs if its the station z. A proper solution is confirmation that all ruins use the /area/ruins pathing and istype from that.

Comment (CitrusGender) [reply code/modules/modular_computers/file_system/programs/alarm.dm]: yes, that's exactly what I did at first. It's up to whatever @ShizCalev wants.

Comment (Jalleo) [reply code/modules/modular_computers/file_system/programs/alarm.dm]: That is a opinion I believe that Shiz doesnt know the station monitors are meant tomonitor the mining areas such as labor camp and general mining too.

Comment (ShizCalev) [reply code/modules/modular_computers/file_system/programs/alarm.dm]:
```
	if(is_station_level(source.z) && !(A.type in GLOB.the_station_areas))
		return
	else if(!is_mining_level(source.z) || istype(A, /area/ruin))
		return
```
my b., that'll fix it for station side ruins too.

**Comment by CitrusGender** [comment]
Alrighty.

**Review by ShizCalev — APPROVED** [review]

**Pull request merged**